File: java/Samples/Biometrics/VB/AbisSampleVB/Common/GetSubjectDialog.vb
Imports System
Imports System.Windows.Forms
Imports Neurotec.Biometrics
Imports Neurotec.Biometrics.Client

Partial Public Class GetSubjectDialog
	Inherits Form
#Region "Public constructor"
	Public Sub New()
		InitializeComponent()
	End Sub
#End Region

#Region "Public properties"

	Private privateSubject As NSubject
	Public Property Subject() As NSubject
		Get
			Return privateSubject
		End Get
		Set(ByVal value As NSubject)
			privateSubject = value
		End Set
	End Property

	Private privateClient As NBiometricClient
	Public Property Client() As NBiometricClient
		Get
			Return privateClient
		End Get
		Set(ByVal value As NBiometricClient)
			privateClient = value
		End Set
	End Property

#End Region

#Region "Private form events"

	Private Sub BtnOkClick(ByVal sender As Object, ByVal e As EventArgs) Handles btnOk.Click
		Try
			Dim subj As NSubject = New NSubject With {.Id = tbId.Text}

			Dim status As NBiometricStatus = Client.Get(subj)
			If status <> NBiometricStatus.Ok Then
				Utilities.ShowInformation("Failed to retrieve subject. Status: {0}", status)
			Else
				Subject = subj
				DialogResult = System.Windows.Forms.DialogResult.OK
			End If
		Catch ex As Exception
			Utilities.ShowError(ex)
		End Try
	End Sub

	Private Sub GetSubjectDialogLoad(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
		If Client Is Nothing Then
			Throw New ArgumentNullException()
		End If

		Try
			Dim operations As NBiometricOperations = Client.LocalOperations
			If Client.RemoteConnections.Count > 0 Then
				operations = operations Or Client.RemoteConnections(0).Operations
			End If

			If (operations And NBiometricOperations.ListIds) = NBiometricOperations.ListIds Then
				tbId.AutoCompleteCustomSource.AddRange(Client.ListIds())
			End If
		Catch ex As Exception
			Utilities.ShowError(ex)
		End Try
	End Sub

	Private Sub TbIdKeyDown(ByVal sender As Object, ByVal e As KeyEventArgs) Handles tbId.KeyDown
		If e.KeyCode = Keys.Enter Then
			btnOk.PerformClick()
		End If
	End Sub

#End Region
End Class

File: java/Samples/Biometrics/CS/AbisSampleCS/Common/GetSubjectDialog.cs
using System;
using System.Windows.Forms;
using Neurotec.Biometrics;
using Neurotec.Biometrics.Client;

namespace Neurotec.Samples
{
	public partial class GetSubjectDialog : Form
	{
		#region Public constructor
		public GetSubjectDialog()
		{
			InitializeComponent();
		}
		#endregion

		#region Public properties

		public NSubject Subject { get; set; }
		public NBiometricClient Client { get; set; }

		#endregion

		#region Private form events

		private void BtnOkClick(object sender, EventArgs e)
		{
			try
			{
				NSubject subj = new NSubject { Id = tbId.Text };

				NBiometricStatus status = Client.Get(subj);
				if (status != NBiometricStatus.Ok)
				{
					Utilities.ShowInformation("Failed to retrieve subject. Status: {0}", status);
				}
				else
				{
					Subject = subj;
					DialogResult = DialogResult.OK;
				}
			}
			catch (Exception ex)
			{
				Utilities.ShowError(ex);
			}
		}

		private void GetSubjectDialogLoad(object sender, EventArgs e)
		{
			if (Client == null) throw new ArgumentNullException();

			try
			{
				NBiometricOperations operations = Client.LocalOperations;
				if (Client.RemoteConnections.Count > 0)
					operations |= Client.RemoteConnections[0].Operations;

				if ((operations & NBiometricOperations.ListIds) == NBiometricOperations.ListIds)
				{
					tbId.AutoCompleteCustomSource.AddRange(Client.ListIds());
				}
			}
			catch (Exception ex)
			{
				Utilities.ShowError(ex);
			}
		}

		private void TbIdKeyDown(object sender, KeyEventArgs e)
		{
			if (e.KeyCode == Keys.Enter)
			{
				btnOk.PerformClick();
			}
		}

		#endregion
	}
}
